File: Dynamic Programming/longest_increasing_subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        
        int dp[nums.size()];
        
        for(int i=0;i<nums.size();i++)
        {
            dp[i] = 1;
        }
        
        for(int i=1;i<nums.size();i++)
        {
            for(int j=0;j<i;j++)
            {
                if(nums[i] > nums[j])
                {
                    dp[i] = max(dp[i],1+dp[j]);
                }
            } 
        }
        
        int maxi = 0;
        for(int i=0;i<nums.size();i++)
        {
            if(dp[i]>=maxi)
            {
                maxi = dp[i];
            }
        }
        
        return maxi;
        
        
}};

File: Dynamic Programming/longest_increasing_subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        
        #initialize the dp array
        dp = [1]*len(nums)
        
        
        #compare from element 0 until current element
        #check if its lesser than the current elements
        for i in range(1,len(nums)):
            for j in range(0,i):
                
                #nums[i] is the current element here
                if nums[i] > nums[j]:
                    dp[i] = max(dp[i],1+dp[j])      #update value if condition is true
                    
        return max(dp)
